consumer/pdatautil/pdatautil.go

```diff
@@ -32,11 +27,8 @@ import (
 //
 // This is a temporary function that will be removed when the new internal pdata.Metrics will be finalized.
 func MetricsToMetricsData(md pdata.Metrics) []consumerdata.MetricsData {
-	if cmd, ok := md.InternalOpaque.([]consumerdata.MetricsData); ok {
-		return cmd
-	}
-	if ims, ok := md.InternalOpaque.(data.MetricData); ok {
-		return internaldata.MetricsToOC(ims)
+	if imd, ok := md.InternalOpaque.(data.MetricData); ok {
```

Comment:
Do we still need to keep `InternalOpaque` in pdata.Metrics as `interface{}` or we can replace it by a concrete pointer to `data.MetricData`?

Reply:
That's what I am doing in the other PR that I split this one. It is a too large change, this is an intermediate step so I can also update contrib easier

Reply:
And I cannot use directly data because of circular dependencies